### PR TITLE
Add Android Robolectric testing example

### PR DESCRIPTION
**Pull request by Bencodes**
Adding an example app showcasing how to create Robolectric tests with Bazel.

### CODE REVIEW

File: android/robolectric-testing/WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+load("@build_bazel_rules_android//android:rules.bzl", "android_sdk_repository")
+
+android_sdk_repository(
+    name = "androidsdk",
+)
+
+## Kotlin
+## Note: For Hilt to work correctly, we are using a pre-release branch being worked on by mauriciogg
```
Comment (ahumesky) [inline android/robolectric-testing/WORKSPACE:45]: is this correct? the URL below looks like a regular release / release candidate ("https://github.com/bazelbuild/rules_kotlin/releases/download/v1.7.0-RC-3/rules_kotlin_release.tgz")

Comment (Bencodes) [reply android/robolectric-testing/WORKSPACE]: Didn't mean to push that bit. Removed the comment for now.

File: android/robolectric-testing/app/src/test/java/com/example/android/bazel/AndroidHelperTest.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.example.android.bazel
+
+import com.google.common.truth.Truth
+import org.junit.Test
+
+class AndroidHelperTest {
```
Comment (ahumesky) [inline android/robolectric-testing/app/src/test/java/com/example/android/bazel/AndroidHelperTest.kt:6]: Could we incorporate some android-related classes in the test? i.e. to show that the robolectric parts are working?

e.g. could we copy/paste the robolectric example? https://robolectric.org/writing-a-test/

Comment (Bencodes) [reply android/robolectric-testing/app/src/test/java/com/example/android/bazel/AndroidHelperTest.kt:6]: Updating this now.

**Comment by Bencodes** [comment]
@ahumesky pushed a better test example that mostly-copies the one from the Robolectric getting started page.

**Comment by ahumesky** [comment]
thank you, looks great!

**Review by ahumesky — APPROVED** [review]

**Pull request merged**